### PR TITLE
[2.0.x] Fix SWITCHING_NOZZLE compile error

### PR DESCRIPTION
**Pull request by scott0122**
#11632 SWITCHING_NOZZLE  compile error

### CODE REVIEW

File: Marlin/src/module/tool_change.cpp [hunk 1]
```diff
@@ -424,6 +416,13 @@ void tool_change(const uint8_t tmp_extruder, const float fr_mm_s/*=0.0*/, bool n
             }
           #endif
 
+          #if ENABLED(SWITCHING_NOZZLE)
+            // Always raise by at least 1 to avoid workpiece
+            current_position[Z_AXIS] += ( (zdiff > 0.0 ? 0.0 : -zdiff) + settings.switching_nozzle_raise_hotend );
```
Comment (thinkyhead) [inline Marlin/src/module/tool_change.cpp]: No such thing as `settings.switching_nozzle_raise_hotend` in Marlin at this time.

**Review by thinkyhead — CHANGES_REQUESTED** [review]
Some changes needed. But I can do that…

**Review by Chuangxiang3D — APPROVED** [review]
All's good now.

**Review by thinkyhead — APPROVED** [review]

**Comment by scott0122** [comment]
Oops...
line 436 to 439 should delete since line 429 already add zdiff

old zdiff code
const float zdiff = hotend_offset[Z_AXIS][**active_extruder**] - hotend_offset[Z_AXIS][**tmp_extruder**];

new zdiff code
zdiff = hotend_offset[Z_AXIS][**tmp_extruder**] - hotend_offset[Z_AXIS][**active_extruder**];

so zdiff is reversed

line 429
`          current_position[Z_AXIS] += zdiff;`

line 436 to 439
`        #if ENABLED(SWITCHING_NOZZLE)`
`          // The newly-selected extruder Z is actually at...`
`          current_position[Z_AXIS] -= zdiff;`
`        #endif`

**Comment by thinkyhead** [comment]
Everything should now be consistent @scott0122 — Please confirm.

**Pull request merged**